test/api.js

```diff
@@ -761,37 +835,187 @@ describe('/v1', function() {
+
+        it('should be able to reduce scopes', function() {
+          return newToken({
+            access_type: 'offline',
+            scope: 'foo bar:baz'
+          }).then(function(res) {
+            assert.equal(res.statusCode, 200);
+            assert.equal(res.result.scope, 'foo bar:baz');
+            return Server.api.post({
+              url: '/token',
+              payload: {
+                client_id: clientId,
+                client_secret: secret,
+                grant_type: 'refresh_token',
+                refresh_token: res.result.refresh_token,
+                scope: 'foo'
+              }
+            });
+          }).then(function(res) {
+            assert.equal(res.statusCode, 200);
+            assert.equal(res.result.scope, 'foo');
+          });
+        });
+
+        it('should not expand scopes', function() {
+          return newToken({
+            access_type: 'offline',
+            scope: 'foo bar:baz'
+          }).then(function(res) {
+            assert.equal(res.statusCode, 200);
+            assert.equal(res.result.scope, 'foo bar:baz');
+            return Server.api.post({
+              url: '/token',
+              payload: {
+                client_id: clientId,
+                client_secret: secret,
+                grant_type: 'refresh_token',
+                refresh_token: res.result.refresh_token,
+                scope: 'foo quux'
+              }
+            });
+          }).then(function(res) {
+            assert.equal(res.statusCode, 400);
+            assert.equal(res.result.errno, 114);
+          });
+        });
+
+      });
+
+      describe('?ttl', function() {
+
+        it('should reduce the expires_in of the access_token', function() {
+          return newToken({ access_type: 'offline' }).then(function(res) {
+            assert.equal(res.statusCode, 200);
+            return Server.api.post({
+              url: '/token',
+              payload: {
+                client_id: clientId,
+                client_secret: secret,
+                grant_type: 'refresh_token',
+                refresh_token: res.result.refresh_token,
+                ttl: 60
+              }
+            });
+          }).then(function(res) {
+            assert.equal(res.statusCode, 200);
+            assert(res.result.expires_in <= 60);
+          });
+        });
+
+        it('should not exceed the maximum', function() {
+          return newToken({ access_type: 'offline' }).then(function(res) {
+            assert.equal(res.statusCode, 200);
+            return Server.api.post({
+              url: '/token',
+              payload: {
+                client_id: clientId,
+                client_secret: secret,
+                grant_type: 'refresh_token',
+                refresh_token: res.result.refresh_token,
+                ttl: MAX_TTL_S * 100
+              }
+            });
+          }).then(function(res) {
+            assertRequestParam(res.result, 'ttl');
```

Comment:
n/m, I see the helper function does this internally.  That's not clear from the name, but I'll file a follow-up bug about it.